File: Programmers/17681_비밀지도/Programmers17681.py
def addZeroToLeft(n, x):
    return '0'*(n-len(x)) + x

def solution(n, arr1, arr2):
    answer = [bin(x1|x2)[2:] for x1,x2 in zip(arr1,arr2)]
    answer = [addZeroToLeft(n, x) for x in answer]
    answer = [x.replace('1','#').replace('0',' ') for x in answer]
    return answer

solution(5, [9, 20, 28, 18, 11], [30, 1, 21, 17, 28])

File: Programmers/17681_비밀지도/Programmers17681.java
public class Programmers17681 {
    public static void main(String args[]){
        System.out.println(new Solution()
                .solution(5, new int[]{9, 20, 28, 18, 11}, new int[]{30, 1, 21, 17, 28}));
    }
}

class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        for (int i = 0; i < n; i++) {
            String bin = Integer.toBinaryString(arr1[i] | arr2[i]);
            bin = (new String(new char[n-bin.length()]).replace("\0", "0")) + bin;
            answer[i] = bin.replace("1", "#").replace("0", " ");
        }
        return answer;
    }
}
